packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1445,6 +1445,9 @@ Directory createTemporaryFlutterSdk(FileSystem fileSystem, Directory realFlutter
       .childFile('pubspec.yaml')
       ..createSync(recursive: true);
     final PubspecYaml pubspecYaml = pubspecsByName[flutterPackage];
+    if (pubspecYaml == null) {
+      continue;
+    }
```

Comment:
This is an improvement. 

Should we print some kind of warning here that the developer has an unexpected folder in `packages`, and it's being ignored?

Reply:
Good call. Done.